### PR TITLE
feat(arc-531): add GET translations/nl.json

### PR DESCRIPTION
**Pull request by bartnaessens**
- (public) translations module: `GET translations/nl.json`. Uses caching.

### CODE REVIEW

File: src/modules/translations/services/translations.service.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		@Inject(CACHE_MANAGER) private cacheManager: Cache
+	) {}
+
+	public async getTranslations(): Promise<Translations> {
+		let translations: Translations = await this.cacheManager.get('translations');
+		if (!translations) {
+			const response = await this.dataService.execute(GET_SITE_VARIABLES_BY_NAME, {
+				name: `translations-frontend`,
+			});
+			translations = get(response, 'data.cms_site_variables[0]');
```
Comment (bertyhell) [inline src/modules/translations/services/translations.service.ts:25]: lets throw a 404 not found if the value is undefined with message: "No translations have been set in the database"

File: src/modules/translations/services/translations.service.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	constructor(
+		private dataService: DataService,
+		@Inject(CACHE_MANAGER) private cacheManager: Cache
+	) {}
+
+	public async getTranslations(): Promise<Translations> {
+		let translations: Translations = await this.cacheManager.get('translations');
+		if (!translations) {
+			const response = await this.dataService.execute(GET_SITE_VARIABLES_BY_NAME, {
+				name: `translations-frontend`,
```
Comment (bertyhell) [inline src/modules/translations/services/translations.service.ts:23]: Keep the same naming convention as with the TOS:
```suggestion
				name: `TRANSLATIONS_FRONTEND`,
```

image

File: src/modules/translations/services/translations.service.ts (hunk 1 above)
Comment (bertyhell) [reply src/modules/translations/services/translations.service.ts:25]: Wouldn't we want to return only the values?
```suggestion
			translations = get(response, 'data.cms_site_variables[0].value');
```

Comment (bartnaessens) [reply src/modules/translations/services/translations.service.ts:25]: updated

File: src/modules/translations/services/translations.service.ts (hunk 2 above)
Comment (bartnaessens) [reply src/modules/translations/services/translations.service.ts:23]: updated

File: src/modules/translations/services/queries.gql.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const GET_SITE_VARIABLES_BY_NAME = `
+	query getSiteVariableByName($name: String!) {
+		cms_site_variables(where: {name: {_eq: $name}}) {
+			name
```
Comment (bertyhell) [inline src/modules/translations/services/queries.gql.ts:4]: no need to fetch name here

File: src/modules/translations/types.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export interface Translations {
+	name: string;
+	value: Record<string, string>;
+}
```
Comment (bertyhell) [inline src/modules/translations/types.ts:4]:
```suggestion
export type Translations = Record<string, string>;
```

**Review by bertyhell — CHANGES_REQUESTED** [review]

**Review by bertyhell — APPROVED** [review]

**Pull request merged**